nsq/conn.py

```diff
@@ -300,15 +300,19 @@ def upgrade_to_tls(self, options=None):
         #
         # first remove the event handler for the currently open socket
         # so that when we add the socket to the new SSLIOStream below,
-        # it can re-add the appropriate event handlers.
+        # it can re-add the appropriate event handlers. Default to TLSv1.2
+        # unless ssl_version is set otherwise.
         self.io_loop.remove_handler(self.socket.fileno())
 
         opts = {
             'cert_reqs': ssl.CERT_REQUIRED,
             'ca_certs': default_ca_certs()
         }
         opts.update(options or {})
-        self.socket = ssl.wrap_socket(self.socket, ssl_version=ssl.PROTOCOL_TLSv1,
+        if not opts.get('ssl_version'):
+            opts['ssl_version'] = ssl.PROTOCOL_TLSv1_2
```

Comment:
there's a slightly more elegant way to do this - if you include `ssl_version` in the `opts` dict initially (just above, next to `ca_certs`), then `opts.update(options)` can override it

(sorry for not seeing this earlier)

Reply:
Ah, yes. I like that much better. I'll have that up in a sec.